src/SWP/Bundle/CoreBundle/Service/RevisionAwareContainerService.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Superdesk Web Publisher Core Bundle.
+ *
+ * Copyright 2015 Sourcefabric z.u. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2015 Sourcefabric z.ú
+ * @license http://www.superdesk.org/license
+ */
+
+namespace SWP\Bundle\CoreBundle\Service;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\ORM\EntityManagerInterface;
+use SWP\Bundle\TemplatesSystemBundle\Service\ContainerService;
+use SWP\Bundle\TemplatesSystemBundle\Service\ContainerServiceInterface;
+use SWP\Component\Revision\Model\RevisionInterface;
+use SWP\Component\Revision\RevisionAwareInterface;
+use SWP\Component\TemplatesSystem\Gimme\Model\ContainerInterface;
+use SWP\Component\TemplatesSystem\Gimme\Model\ContainerWidgetInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface as ServiceContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Class RendererService.
+ */
+class RevisionAwareContainerService extends ContainerService implements ContainerServiceInterface
+{
+    /**
+     * RevisionAwareContainerService constructor.
+     *
+     * @param EntityManagerInterface    $entityManager
+     * @param EventDispatcherInterface  $eventDispatcher
+     * @param ServiceContainerInterface $serviceContainer
+     */
+    public function __construct(
+        EntityManagerInterface $entityManager,
+        EventDispatcherInterface $eventDispatcher,
+        ServiceContainerInterface $serviceContainer
+    ) {
+        parent::__construct($entityManager, $eventDispatcher, $serviceContainer);
```

Comment:
what about creating abstract container and put there methods that will be re-used by other implementations, together with the construct? it will reduce the code size in container services